### DISCUSSION TITLE
Error with mesh appearing on the file

### DISCUSSION THREAD

**Discussion by Slykkey** [sofa-framework/sofa#4488] [BeamAdapter] [answered]
Hello,

I have made a post about this problem earlier last month. I tried to run the same code on master version with BeamAdapter 23.12 which uses Rodxxxsection. I have used the mesh 
![300489188-a40695eb-aee4-45c4-b776-fd89adfea78f](https://github.com/sofa-framework/sofa/assets/69535855/d47c3c1a-a3cf-43fd-b374-b7412732cf53)
and my output does not collapse but it starts off straight and unstable like 
![Screenshot from 2024-02-08 20-54-09](https://github.com/sofa-framework/sofa/assets/69535855/b193809f-a5f4-471b-b06e-a84f2003c0d6)
and then towards the end I can see the shape 
![Screenshot from 2024-02-08 20-54-26](https://github.com/sofa-framework/sofa/assets/69535855/f80eda25-b8b5-45db-8ea3-ef92bd58b1d4)
Can I ask if this is because I have to change the axis while I design the mesh? It feels like the straight part comes out and then the shape of the catheter I designed actually comes at the end. Please let me know if you need anything more

**Comment by epernod** [comment, marked as answer]
Hi @Slykkey ,

The order of the sections maters when linked in the WireRestShape, it is modelling the tool from the handle to the tip. So the last section linked will be the first one to get out. 
```<WireRestShape template="Rigid3d" name="GC_RestShape" printLog="false" wireMaterials="@StraightSection @MeshTipSection" />```

Best regards

**Reply by Slykkey** [reply]
Thanks @epernod ,

I was able to make the shape work, but I had to design it a particular way to make it appear on the simulation. Do you have any information on how exactly the mesh should be design to run a smooth simulation? Thanks for the answer, appreciate it.

**Reply by epernod** [reply]
You need to mesh it as a continuous set of lines. (like the skeleton of your tool) 
Then for smooth simulation, you should have enough edges to perform a good collision and accurate physical modelling. After you need to set the good physical properties on it.

**Reply by Slykkey** [reply]
Thanks @epernod ,

I do have a few questions about it. What does the number of edges change in the code? I have a working collision model with good collision response. Also the value of radius in WireBeamInterpolation denotes what radius? Is it the radius of the instrument or is it a different value? I get different responses in the simulation when I change this radius value. If you can refer me to an example regarding this, it would be great. Thanks a lot!

**Comment by athishsm** [comment]
Hi,

I need a help with an issue similar to this. I have a centreline (as .msh) of a spring (volume mesh shown below).
![image](https://github.com/user-attachments/assets/fbf020fa-2ae8-4945-8584-96d985146269)

I need to model this centreline in SOFA with Beam elements and assign metallic material properties. Could you please let me know how to import my own beam mesh inside SOFA?
